File: src/main/java/com/jss/osiris/modules/miscellaneous/service/SalesReclamationOriginService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.SalesReclamationOrigin;

public interface SalesReclamationOriginService {

        public List<SalesReclamationOrigin> getReclamationOrigins();

        public SalesReclamationOrigin addOrUpdateReclamationOrigin(SalesReclamationOrigin salesReclamationOrigin);
}

File: client/src/app/modules/tiers/services/sales.reclamation.origin.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { SalesReclamationOrigin } from '../../miscellaneous/model/SalesReclamationOrigin';

@Injectable({
  providedIn: 'root'
})
export class SalesReclamationOriginService extends AppRestService<SalesReclamationOrigin>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getReclamationOrigins() {
    return this.getListCached(new HttpParams(), "sales-reclamation-origins");
  }

   addOrUpdateReclamationOrigin(salesReclamationOrigin: SalesReclamationOrigin) {
    return this.addOrUpdate(new HttpParams(), "sales-reclamation-origin", salesReclamationOrigin, "Enregistré", "Erreur lors de l'enregistrement");
  }
}
